busbus/web.py

```diff
@@ -0,0 +1,154 @@
+import busbus
+from busbus.entity import BaseEntityJSONEncoder
+from busbus.provider import ProviderBase
+
+import cherrypy
+import collections
+import itertools
+import math
+import types
+
+
+def json_handler(*args, **kwargs):
+    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)
+    return BaseEntityJSONEncoder().encode(value).encode('utf-8')
+cherrypy.config['tools.json_out.handler'] = json_handler
+
+
+EXPAND_TYPES = {
+    'providers': ProviderBase,
+    'agencies': busbus.Agency,
+    'stops': busbus.Stop,
+    'routes': busbus.Route,
+    'arrivals': busbus.Arrival,
+}
+
+
+def unexpand_init(result, to_expand):
+    return ({attr: unexpand(value, to_expand)
+             for attr, value in dict(obj).items()}
+            for obj in result)
+
+
+def unexpand(obj, to_expand):
+    for name, cls in EXPAND_TYPES.items():
+        if isinstance(obj, cls):
+            if name not in to_expand:
+                return {'id': obj.id}
+            else:
+                return {attr: unexpand(value, to_expand)
+                        for attr, value in dict(obj).items()}
+    if isinstance(obj, dict):
+        return {attr: unexpand(value, to_expand)
+                for attr, value in obj.items()}
+    if isinstance(obj, (list, tuple, collections.Iterator)):
+        return (unexpand(value, to_expand) for value in obj)
+    return obj
+
+
+class APIError(Exception):
+
+    def __init__(self, msg, error_code=500):
+        self.msg = msg
+        self.error_code = error_code
+
+
+class Engine(busbus.Engine):
+
+    def __init__(self, *args, **kwargs):
+        # perhaps fix this to use a decorator somehow?
```

Comment:
Making this work with a decorator is probably not within the scope I want for this pull request